### PR TITLE
feat: SSE live-update bus, mapping, and stream endpoint (PR 1/3)

### LINKED ISSUE #446: Add event-day live update streams for check-in, seating, orders, and delivery
## Context

Champagne Festival does not appear to have an SSE/live-update layer yet. Event-day operations would benefit from live updates because check-in, seating, order status, and champagne delivery can change while multiple volunteers/admins are looking at the same screens.

Use the existing patterns from Daynest and Worktime, but keep the design event-day focused.

## Scope

- Add a scoped live-update endpoint for operational event-day views.
- Prefer notify-then-pull: stream small invalidation events and let clients refetch the relevant query/read model.
- Define event names and payload contracts for:
  - registration/check-in changes
  - seating/table assignment changes
  - order changes
  - champagne delivery quantity changes
- Scope streams by role and event/edition where appropriate, so volunteers only receive operational updates they are allowed to see.
- Add frontend hooks/providers that integrate with TanStack Query invalidation when live events arrive.
- Add tests for auth, role scoping, stream headers, keepalive behavior, and mutation-triggered broadcasts.
- Add MSW/test support for deterministic frontend live-update scenarios and screenshots.

## Acceptance criteria

- Event-day admin/volunteer screens can refresh from live invalidation events without manual reload.
- Stream payloads stay small and do not leak unauthorized data.
- Tests cover auth/role scoping, event delivery, keepalives, and query invalidation.
- Documentation explains reconnect behavior and missed-event recovery.
- MSW or test utilities can simulate live events for frontend tests and screenshot workflows.

### PR DESCRIPTION
**Pull request by tjorim**
## Summary

Adds the backend infrastructure for event-day live updates via Server-Sent Events. No mutation routes are wired yet — that is PR 2/3.

- **`app/live/events.py`** — frozen `LiveEvent` + `LiveScope` dataclasses with `to_sse_data()` serialisation
- **`app/live/bus.py`** — `LiveBus` in-process asyncio pub/sub hub with bounded per-subscriber queues (slow-consumer protection: drop + warning log)
- **`app/live/mapping.py`** — authoritative mapping from domain actions to TanStack Query keys; `backend/app/live/mapping.py` is the source of truth for which frontend keys each mutation invalidates
- **`app/routers/live.py`** — `GET /api/live/stream` SSE endpoint requiring `volunteer`/`admin` role; streams `event: invalidate` frames + keepalive comments every 15 s; scope-filterable via `?edition_id=` and `?event_id=`

## Architecture decisions

- **In-process asyncio pub/sub** — no Redis needed for the current single-Uvicorn deployment; `LiveBus` interface is the only callsite so a Redis backend can be swapped in later without touching routers
- **`get_heartbeat_interval` FastAPI dependency** — injectable in tests without monkeypatching
- **Generator-direct testing** — `httpx.ASGITransport` buffers the full response body before returning, which hangs on infinite SSE streams; auth tests use httpx, SSE content tests call `_sse_generator()` directly

## Test plan

- [x] 39 tests pass in 0.35 s (no database required)
- [x] `test_live_bus.py` — subscribe/publish, multi-subscriber, slow-consumer drop, unsubscribe cleanup, exception propagation
- [x] `test_live_mapping.py` — topic/action/scope/keys for all 5 mapping functions + SSE serialisation format
- [x] `test_live_endpoint.py` — 401/403 auth, SSE headers, ready event, keepalive, event delivery, edition_id + event_id scope filtering, subscriber cleanup on generator close
- [x] `ruff check` clean on all new files

## Infra note

Production Caddy needs `flush_interval -1` for `/api/live/stream` — tracked in tjorim/apps#77.

Closes #446 (partial — 1/3)

### CODE REVIEW

File: backend/app/routers/live.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                continue
+
+            # Scope filtering: skip events that don't match the requested scope.
+            # A None scope value on the event means "applies to all".
+            if edition_id is not None and event.scope.edition_id not in (None, edition_id):
+                continue
+            if event_id is not None and event.scope.event_id not in (None, event_id):
+                continue
+
+            yield event.to_sse_data()
```
[the remark this answers is not in the document]
Comment (tjorim) [reply backend/app/routers/live.py:71]: Fixed in feat/live-update-streams-pr3 (d9c604e) — carrying the fix here rather than amending PR 1 since the stack is already open for review.

The generator now tracks `last_sent` via `loop.time()` and recalculates the remaining wait on every iteration, so `asyncio.wait_for()` always times out within `heartbeat_interval` seconds regardless of how many filtered events are consumed. Keepalive is also emitted immediately at the top of the loop when `remaining <= 0`, covering the edge case where filtered-event processing itself eats into the budget.

Added a regression test (`test_sse_generator_keepalive_not_starved_by_filtered_events`) that floods 20 non-matching events and asserts `: keepalive` is the next chunk — all 40 tests pass.

**Comment by tjorim** [comment]
@coderabbitai review
